packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -125,6 +108,20 @@ library LibVerifying {
         }
     }
 
+    function withdrawBalance(
+        TaikoData.State storage state,
+        AddressResolver resolver
+    ) public {
+        uint256 balance = state.balances[msg.sender];
+        if (balance == 0) return;
+
+        state.balances[msg.sender] = 0;
```

Comment:
Shall we allow users to leave 1 TKO in their balance when withdrawing? Otherwise after a withdraw, their next reward will be reduced to 1 TKO, is this kind of penalty what we want?

Reply:
Good point, I'll change it a bit.